Remove all available versions of a box

berendt opened the pull request:
This patch introduces a new parameter --all for the remove command of the box plugin.

While using the parameter --all when removing a box will remove all available versions of the box.

Example usage:

```
$ vagrant box list
berendt/ubuntu-14.04-amd64          (virtualbox, 0.2.11)
berendt/ubuntu-14.04-amd64          (virtualbox, 0.2.12)
berendt/ubuntu-14.04-amd64          (virtualbox, 0.2.13)
```

```
$ vagrant box remove berendt/ubuntu-14.04-amd64
You requested to remove the box 'berendt/ubuntu-14.04-amd64' with provider
'virtualbox'. This box has multiple versions. You must
explicitly specify which version you want to remove with
the `--box-version` flag.

Versions:
0.2.13, 0.2.12,  0.2.11
```

Using the --all parameter now it's possible to remove all versions at once.

```
$ vagrant box remove berendt/ubuntu-14.04-amd64 --all
Removing box 'berendt/ubuntu-14.04-amd64' with provider 'virtualbox' in version '0.2.13'...
Removing box 'berendt/ubuntu-14.04-amd64' with provider 'virtualbox' in version '0.2.12'...
Removing box 'berendt/ubuntu-14.04-amd64' with provider 'virtualbox' in version '0.2.11'...
```

Fixes https://github.com/mitchellh/vagrant/issues/3343

sethvargo commented:
Hey @berendt 

I just wanted to check in and see if you had any questions about adding this to the documentation. Let me know if I can help :smile:

berendt commented:
We have our yearly hackweek and I can complete this PR after more than 1 year.. Rebased with the latest updated, next update will add docs...

berendt commented:
Added the missing documentation and updated the commit message. Tested it with the `master` branch and the `--all` parameter is still working like expected.

@sethvargo Should be fine to merge now.

sethvargo commented:
Merged in a8afb40. Thanks and sorry again for the delay! :smile:
